orchestration/api/instances.py

```diff
@@ -99,6 +98,26 @@ def instance_ops(tenant_id=''):
         # If description is not provided, the instance creation should proceed
         logger.info("no user_id provided. Exception [%s]", str(e))
 
+    # action of the instance creator
+    try:
+        action = content['action']
+        if action == '':
+            raise ValueError('Empty action provided')
+        if action == 'opensds.provision-volume':
+            content['parameters']['ip_addr'] = get_config(config_file,
+                                                          'hotpot', 'host')
+            content['parameters']['port'] = get_config(config_file,
+                                                       'hotpot', 'port')
+        else:
+            content['parameters']['ip_addr'] = get_config(config_file,
+                                                          'gelato', 'host')
+            content['parameters']['port'] = get_config(config_file,
+                                                       'gelato', 'port')
+    except Exception as e:
+        err_msg = 'required input action is missing'
```

Comment:
What of the exception is thrown while reading content? err_msg will not be correct